usgs-details.html

```diff
@@ -0,0 +1,99 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <style type="text/css">
+        body {
+            font-family: 'Gill Sans', 'Gill Sans MT', 'Myriad Pro', 'DejaVu Sans Condensed', Helvetica, Arial, sans-serif;
+        }
+
+        .container {
+            width: 700px;
+            height: 700px;
+            background-color: #000000;
+            margin-top: 250px;
+            margin-left: 200px;
+            color: #ffffff;
+        }
+
+        #value {
+            font-size: 36px;
+        }
+
+        #units {
+            font-size: 24px;
+        }
+    </style>
+</head>
+<body>
+    <div class="container">
+        <span id="value"></span>
+        <span id="units"></span>
+    </div>
+    <script type="text/javascript">
+        var SITE_ID = '01474500';
+        var BASE_URL = 'https://waterservices.usgs.gov/nwis/iv/';
+        var REQUEST_URL = BASE_URL + '?format=json' + '&sites=' + SITE_ID;
+
+        // Hat tip https://stackoverflow.com/a/901144/6995854
+        // Utility function to get query parameter
+        function getQueryParameterByName(name, url) {
+            if (!url) url = window.location.href;
+            name = name.replace(/[\[\]]/g, "\\$&");
+            var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
```

Comment:
Not sure whether the code here would be available in all versions of code but I think this code be simplified if it could use `decodeURI(window.location.search).slice(1)` to get the querystring. This seems to work well but the RegEx is a little hard to read.